src/keymap.rs

```diff
@@ -0,0 +1,63 @@
+use glib::translate::*;
+use std::mem;
+use std::ptr;
+use Keymap;
+use KeymapKey;
+
+impl Keymap {
+    pub fn get_entries_for_keycode(
+        &self,
+        hardware_keycode: u32,
+    ) -> Option<(Vec<KeymapKey>, Vec<u32>)> {
+        unsafe {
+            let mut keys = ptr::null_mut();
+            let mut keyvals = ptr::null_mut();
+            let mut n_entries = mem::MaybeUninit::uninit();
+            let ret = from_glib(gdk_sys::gdk_keymap_get_entries_for_keycode(
+                self.to_glib_none().0,
+                hardware_keycode,
+                &mut keys,
+                &mut keyvals,
+                n_entries.as_mut_ptr(),
+            ));
+            if ret {
```

Comment:
Why not return an empty `Vec` if there were no entries? Less wrapping and that's exactly what a return value of `false` signals: no entries.